lib/order.rb

```diff
@@ -1,18 +1,90 @@
+require 'pry'
+require 'csv'
+require 'awesome_print'
+
+
 module Grocery
+  # begin
+  # rescue Grocery::FindError > e
+  # end
+
+  # custom error for the Order.find class method
+  class FindError < ArgumentError
+    def initialize(msg="Error: Order has not been created yet")
+      super
+      # rescue
+    end
+  end
+
   class Order
     attr_reader :id, :products
+    @@all_orders = []
 
     def initialize(id, products)
       @id = id
       @products = products
     end
 
     def total
-      # TODO: implement total
+      subtotal = 0
+      @products.each do |product, price|
+        subtotal += price
+      end
+      total = subtotal + (subtotal * 0.075)
+      return total.round(2)
     end
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
+      products = {product_name => product_price}
+      products.each do |product, price|
+        if @products.include? product
+          return false
+        else
+          @products.merge!({product => price})
```

Comment:
While this certainly works, I think that using `merge!` may be overkill here. It would be simpler to say `@products[product_name] = product_price`.